payments-core/src/main/java/com/stripe/android/paymentsheet/paymentdatacollection/TransformToPaymentMethodCreateParams.kt

```diff
@@ -1,15 +1,41 @@
-package com.stripe.android.paymentsheet.forms
```

Comment:
IIUC we're trying to push all deps to `paymentsheet` module and reverse the dependency later, is it possible to avoid moving classes into `payments-core`?

Reply:
I completely understand where you are coming from.  I think the challenge we face here is that PaymentMethodCreateParams is in core, and this is the glue between the forms and PaymentMethodCreateParams.

I think there are a few options:
1. Make PaymentMethodCreateParams implement an interface and have our own GenericPaymentMethodCreateParams object in PaymentSheet.  The interface would exist in Paymentsheet Module for now, but would move to core eventually.  I started a PR of what this might look like here: https://github.com/stripe/stripe-android/pull/3895 
2. Move PaymentMethodCreateParams to paymentsheet module.  We would just have to move it back into core though when we flip the dependency so that paymentsheet depends on Core.
3. I tried moving BaseAddPaymentMethodFragment, which uses this class into PaymentSheet, but it requires BaseSheetViewModel which depends on PaymentIntent, PaymentMethod, and StripeIntent in core.

I do lean slightly toward leaving this as in the PR.  This code will be very easy to move into the paymentSheet module has it only has dependencies on PaymentMethodCreateParams and it is already a public class (so we don't have to worry about exposing additional dependencies).